--- origsrc/gtk-doc-1.15/gtkdoc-scangobj.in	2010-05-09 15:19:08.000000000 -0500
+++ src/gtk-doc-1.15/gtkdoc-scangobj.in	2010-05-21 11:02:42.452194600 -0500
@@ -104,6 +104,7 @@ my $old_prerequisites_filename = "$OUTPU
 my $new_prerequisites_filename = "$OUTPUT_DIR/$MODULE.prerequisites.new";
 my $old_args_filename = "$OUTPUT_DIR/$MODULE.args";
 my $new_args_filename = "$OUTPUT_DIR/$MODULE.args.new";
+my $exeext = ($^O =~ /MSWin32|cygwin/) ? ".exe" : "";
 
 # write a C program to scan the types
 
@@ -1603,15 +1604,15 @@ system("($command)") == 0 or die "Compil
 print "gtk-doc: Linking scanner\n";
 # FIXME: Can we turn off as-needed for the docs (or better fix it?)
 #$command = "$LD -Wl,--no-as-needed $o_file $LDFLAGS -o $MODULE-scan";
-$command = "$LD $o_file $LDFLAGS -o $MODULE-scan";
+$command = "$LD $o_file $LDFLAGS -o $MODULE-scan$exeext";
 system("($command)") == 0 or die "Linking of scanner failed: $!\n";
 
 print "gtk-doc: Running scanner $MODULE-scan\n";
-system("($RUN ./$MODULE-scan)") == 0 or die "Scan failed: $!\n";
+system("($RUN ./$MODULE-scan$exeext)") == 0 or die "Scan failed: $!\n";
 
 
 if (!defined($ENV{"GTK_DOC_KEEP_INTERMEDIATE"})) {
-  unlink "./$MODULE-scan.c", "./$MODULE-scan.o", "./$MODULE-scan.lo", "./$MODULE-scan";
+  unlink "./$MODULE-scan.c", "./$MODULE-scan.o", "./$MODULE-scan.lo", "./$MODULE-scan$exeext";
 }
 
 &UpdateFileIfChanged ($old_signals_filename, $new_signals_filename, 0);
--- origsrc/gtk-doc-1.15/gtkdoc-scanobj.in	2010-05-09 15:19:17.000000000 -0500
+++ src/gtk-doc-1.15/gtkdoc-scanobj.in	2010-05-21 11:03:14.389021300 -0500
@@ -86,6 +86,7 @@ my $old_hierarchy_filename = "$OUTPUT_DI
 my $new_hierarchy_filename = "$OUTPUT_DIR/$MODULE.hierarchy.new";
 my $old_args_filename = "$OUTPUT_DIR/$MODULE.args";
 my $new_args_filename = "$OUTPUT_DIR/$MODULE.args.new";
+my $exeext = ($^O =~ /MSWin32|cygwin/) ? ".exe" : "";
 
 # write a C program to scan the types
 
@@ -843,14 +844,14 @@ system("($command)") == 0 or die "Compil
 print "gtk-doc: Linking scanner\n";
 # FIXME: Can we turn off as-needed for the docs (or better fix it?)
 #$command = "$LD -Wl,--no-as-needed $o_file $LDFLAGS -o $MODULE-scan";
-$command = "$LD $o_file $LDFLAGS -o $MODULE-scan";
+$command = "$LD $o_file $LDFLAGS -o $MODULE-scan$exeext";
 system("($command)") == 0 or die "Linking of scanner failed: $!\n";
 
 print "gtk-doc: Running scanner $MODULE-scan\n";
-system("($RUN ./$MODULE-scan)") == 0 or die "Scan failed: $!\n";
+system("($RUN ./$MODULE-scan$exeext)") == 0 or die "Scan failed: $!\n";
 
 if (!defined($ENV{"GTK_DOC_KEEP_INTERMEDIATE"})) {
-  unlink "./$MODULE-scan.c", "./$MODULE-scan.o", "./$MODULE-scan.lo", "./$MODULE-scan";
+  unlink "./$MODULE-scan.c", "./$MODULE-scan.o", "./$MODULE-scan.lo", "./$MODULE-scan$exeext";
 }
 
 &UpdateFileIfChanged ($old_signals_filename, $new_signals_filename, 0);
