--- origsrc/gtk-doc-1.21/gtk-doc.flat.make	2014-04-22 06:53:53.000000000 -0500
+++ src/gtk-doc-1.21/gtk-doc.flat.make	2015-01-01 18:49:35.000000000 -0600
@@ -115,6 +115,9 @@ scan-build.stamp: setup-build.stamp $(HF
 	$(GTK_DOC_V_SCAN)_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
+	    case $$i in /*) ;; \
+	    *)	_source_dir="$${_source_dir} --source-dir=$(srcdir)/$$i" ;; \
+	    esac ; \
 	done ; \
 	gtkdoc-scan --module=$(DOC_MODULE) --ignore-headers="$(IGNORE_HFILES)" $${_source_dir} $(SCAN_OPTIONS) $(EXTRA_HFILES)
 	$(GTK_DOC_V_INTROSPECT)if grep -l '^..*$$' $(DOC_MODULE).types > /dev/null 2>&1 ; then \
@@ -168,6 +171,9 @@ sgml-build.stamp: tmpl.stamp $(DOC_MODUL
 	-$(GTK_DOC_V_XML)chmod -R u+w $(srcdir) && _source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
+	    case $$i in /*) ;; \
+	    *)	_source_dir="$${_source_dir} --source-dir=$(srcdir)/$$i" ;; \
+	    esac ; \
 	done ; \
 	gtkdoc-mkdb --module=$(DOC_MODULE) --output-format=xml --expand-content-files="$(expand_content_files)" --main-sgml-file=$(DOC_MAIN_SGML_FILE) $${_source_dir} $(MKDB_OPTIONS)
 	$(AM_V_at)touch sgml-build.stamp
--- origsrc/gtk-doc-1.21/gtk-doc.notmpl-flat.make	2014-04-22 06:53:53.000000000 -0500
+++ src/gtk-doc-1.21/gtk-doc.notmpl-flat.make	2015-01-01 18:49:35.000000000 -0600
@@ -113,6 +113,9 @@ scan-build.stamp: setup-build.stamp $(HF
 	$(GTK_DOC_V_SCAN)_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
+	    case $$i in /*) ;; \
+	    *)	_source_dir="$${_source_dir} --source-dir=$(srcdir)/$$i" ;; \
+	    esac ; \
 	done ; \
 	gtkdoc-scan --module=$(DOC_MODULE) --ignore-headers="$(IGNORE_HFILES)" $${_source_dir} $(SCAN_OPTIONS) $(EXTRA_HFILES)
 	$(GTK_DOC_V_INTROSPECT)if grep -l '^..*$$' $(DOC_MODULE).types > /dev/null 2>&1 ; then \
@@ -145,6 +148,9 @@ sgml-build.stamp: setup-build.stamp $(DO
 	$(GTK_DOC_V_XML)_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
+	    case $$i in /*) ;; \
+	    *)	_source_dir="$${_source_dir} --source-dir=$(srcdir)/$$i" ;; \
+	    esac ; \
 	done ; \
 	gtkdoc-mkdb --module=$(DOC_MODULE) --output-format=xml --expand-content-files="$(expand_content_files)" --main-sgml-file=$(DOC_MAIN_SGML_FILE) $${_source_dir} $(MKDB_OPTIONS)
 	$(AM_V_at)touch sgml-build.stamp
--- origsrc/gtk-doc-1.21/gtk-doc.notmpl.make	2014-02-14 02:27:47.000000000 -0600
+++ src/gtk-doc-1.21/gtk-doc.notmpl.make	2015-01-01 18:49:35.000000000 -0600
@@ -113,6 +113,9 @@ scan-build.stamp: setup-build.stamp $(HF
 	$(GTK_DOC_V_SCAN)_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
+	    case $$i in /*) ;; \
+	    *)	_source_dir="$${_source_dir} --source-dir=$(srcdir)/$$i" ;; \
+	    esac ; \
 	done ; \
 	gtkdoc-scan --module=$(DOC_MODULE) --ignore-headers="$(IGNORE_HFILES)" $${_source_dir} $(SCAN_OPTIONS) $(EXTRA_HFILES)
 	$(GTK_DOC_V_INTROSPECT)if grep -l '^..*$$' $(DOC_MODULE).types > /dev/null 2>&1 ; then \
@@ -145,6 +148,9 @@ sgml-build.stamp: setup-build.stamp $(DO
 	$(GTK_DOC_V_XML)_source_dir='' ; \
 	for i in $(DOC_SOURCE_DIR) ; do \
 	    _source_dir="$${_source_dir} --source-dir=$$i" ; \
+	    case $$i in /*) ;; \
+	    *)	_source_dir="$${_source_dir} --source-dir=$(srcdir)/$$i" ;; \
+	    esac ; \
 	done ; \
 	gtkdoc-mkdb --module=$(DOC_MODULE) --output-format=xml --expand-content-files="$(expand_content_files)" --main-sgml-file=$(DOC_MAIN_SGML_FILE) $${_source_dir} $(MKDB_OPTIONS)
 	$(AM_V_at)touch sgml-build.stamp
